@here/harp-mapview-decoder/lib/TileLoader.ts

```diff
@@ -111,6 +111,7 @@ export class TileLoader {
                 this.ensureLoadingStarted();
                 return this.donePromise!;
         }
+        assertNever(this.state);
```

Comment:
This is not needed if we trust TypeScript.
Typescript already checks if all paths return value.

Experiment: comment `TileLoaderState.Failed` case and see that typescript doesn't even compile this.

If we don't trust `this.state` then this would be viable check.

Reply:
Thanks for pointing out, yes that is correct. I will remove.